components/raftstore/src/store/peer_storage.rs

```diff
@@ -597,24 +621,29 @@ where
         for r in destroy_regions {
             write_peer_state(kv_wb, r, PeerState::Tombstone, None)?;
         }
-        write_peer_state(kv_wb, &region, PeerState::Applying, None)?;
 
-        let last_index = snap.get_metadata().get_index();
+        // Witness snapshot is applied atomically as no async applying operation to
+        // region worker, so no need to set the peer state
```

Comment:
It doesn't need to set the state to `Applying`, but it needs to persist the region info.